scarlet/component.py

```diff
@@ -155,6 +167,10 @@ def get_model(self, sed=None, morph=None):
             return self._sed[:, None, None] * self._morph[None, :, :]
         raise ValueError("You need to supply `sed` and `morph` or neither")
 
+    def update_gradients(self, grad_sed, grad_morph):
```

Comment:
I don't get this function: it has a fixed number of specified parameters. isn't the idea of this PR to allow for an arbitrary arrangement of parameters.

Reply:
This is because most components will just have `sed` and `morph`. For sources with more complicated models `get_model` is actually rewritten by an inherited class (as it must be, because it will contain a different model). I'll post an example on slack.